src/pymor/bindings/dunegdt.py

```diff
@@ -0,0 +1,189 @@
+                            pvd_file.write('  <Collection>\n')
+                            timestep = 0
+                            for f_name in data:
+                                pvd_file.write('    <DataSet timestep="{}" group="" part="0" file="{}"/>\n'.format(
+                                    timestep, f_name))
+                                timestep += 1
+                            pvd_file.write('  </Collection>\n')
+                            pvd_file.write('</VTKFile>\n')
+                        subprocess.call(['paraview', pvd_filename])
+                        for f_name in data:
+                            to_clean_up.append(f_name)
+                        to_clean_up.append(pvd_filename)
+
+            for f_name in to_clean_up:
+                os.remove(f_name)
+
+
+    class DuneGDTK3dVisualizer(ImmutableObject):
+        """Visualize a dune-gdt discrete function within a jupyter notebook using K3d.
+
+        Parameters
+        ----------
+        space
+            The dune-gdt space for which we want to visualize DOF vectors.
+        """
+
+        def __init__(self, space, name='STATE'):
+            # self.dune_vec = CommonDenseVectorDouble(space.num_DoFs)
+            # self.np_view = np.array(self.dune_vec, copy=False)
+            # self.discrete_function = make_discrete_function(space, self.dune_vec, name)
+            self.__auto_init(locals())
+
+        def visualize(self, U, m):
+
+            def visualize_single(u, filename):
+                assert isinstance(u, DuneXTVector)
+                df = DiscreteFunction(self.space, u.impl, self.name)
+                df.visualize(filename)
+
+            prefix = os.path.join(tempfile.gettempdir(), next(tempfile._get_candidate_names()))
+            suffix = 'vt{}'.format('k' if self.space.dimDomain == 1 else 'u')
+            if isinstance(U, ListVectorArray):
+                if len(U) == 0:
+                    return
+                for i in range(len(U)):
+                    visualize_single(U._list[i], f'{prefix}_{i}')
+                if len(U) == 1:
+                    filename = f'{prefix}_{i}.{suffix}'
+                else:
+                    with open(f'{prefix}.pvd', 'w') as pvd_file:
+                        pvd_file.write('<?xml version=\'1.0\'?>\n')
+                        pvd_file.write('<VTKFile type=\'Collection\' version=\'0.1\'>\n')
+                        pvd_file.write('<Collection>\n')
+                        for i in range(len(U)):
+                            pvd_file.write(f'<DataSet timestep=\'{i}\' part=\'1\' name=\'{self.name}\' file=\'{prefix}_{i}.{suffix}\'/>\n')
+                        pvd_file.write('</Collection>\n')
+                        pvd_file.write('</VTKFile>\n') 
+                    filename = f'{prefix}.pvd'
+            else:
+                raise NotImplementedError
+            _ = k3d_plot(filename, color_attribute_name=self.name)
```

Comment:
Can you return the plot object here to allow modification in jupyter?

Reply:
done in #1381